c7n/filters/core.py

```diff
@@ -760,3 +760,17 @@ def get_resource_value(self, resource):
         if capture is None:  # regex didn't capture anything
             return None
         return capture.group(1)
+
+
+class StateTransitionFilter(Filter):
+    valid_origin_states = ()
+
+    def filter_resource_state(self, resources, event=None):
+        state_key = self.manager.get_model().state_key
+        states = self.valid_origin_states
+        orig_length = len(resources)
+        results = [r for r in resources if r[state_key] in states]
+        self.log.info("%s %d of %d resources" % (
```

Comment:
the log message here isn't very descriptive, add a line about implicitly filtered %s of %s resources